include/deal.II/lac/vector_operations_internal.h

```diff
@@ -2472,32 +2478,22 @@ namespace internal
         const std::shared_ptr<::dealii::parallel::internal::TBBPartitioner> &,
         const size_type size,
         const ::dealii::MemorySpace::
-          MemorySpaceData<Number, ::dealii::MemorySpace::CUDA> &data)
-      {
-        Number *    result_device;
-        cudaError_t error_code = cudaMalloc(&result_device, sizeof(Number));
-        AssertCuda(error_code);
-        error_code = cudaMemset(result_device, 0, sizeof(Number));
-
-        const int n_blocks = 1 + size / (chunk_size * block_size);
-        ::dealii::LinearAlgebra::CUDAWrappers::kernel::reduction<
-          Number,
-          ::dealii::LinearAlgebra::CUDAWrappers::kernel::ElemSum<Number>>
-          <<<dim3(n_blocks, 1), dim3(block_size)>>>(result_device,
-                                                    data.values.data(),
-                                                    size);
-
-        // Copy the result back to the host
+          MemorySpaceData<Number, ::dealii::MemorySpace::Default> &data)
+      {
         Number result;
-        error_code = cudaMemcpy(&result,
-                                result_device,
-                                sizeof(Number),
-                                cudaMemcpyDeviceToHost);
-        AssertCuda(error_code);
-        // Free the memory on the device
-        error_code = cudaFree(result_device);
-        AssertCuda(error_code);
 
+        auto exec = typename ::dealii::MemorySpace::Default::kokkos_space::
+          execution_space{};
+        Kokkos::parallel_reduce(
+          Kokkos::RangePolicy<
+            ::dealii::MemorySpace::Default::kokkos_space::execution_space>(
+            exec, 0, size),
+          KOKKOS_LAMBDA(size_type i, Number & update) {
+            update += data.values(i);
+          },
+          result);
```

Comment:
The name of the function is `mean_value`, shouldn't we divide the `result` by the size?